kite-data/kite-data-core/src/main/java/org/kitesdk/data/DatasetException.java

```diff
@@ -46,4 +46,11 @@ public DatasetException(Throwable t) {
     super(t);
   }
 
+  protected static String format(String message, Object... args) {
+    String[] argStrings = new String[args.length];
+    for (int i = 0; i < args.length; i += 1) {
+      argStrings[i] = String.valueOf(args[i]);
+    }
+    return String.format(String.valueOf(message), (Object[]) argStrings);
```

Comment:
First, I like being a little more careful with error messages so that any null values end up being "null" rather than causing a different exception and accidentally suppressing the one that matters. That's not too likely, though.

The other reason is to try to match the style of preconditions, which always use `%s`. It's annoying to need to remember whether the formatter for a method uses `{}`, `%s`, or allows `%d` etc. Plus that turns into a very bad thing when you get it wrong because it's a runtime bug and you usually lose key error information.